corehq/apps/users/models.py

```diff
@@ -2124,6 +2124,18 @@ def get_sql_location(self, domain):
         if loc_id:
             return SQLLocation.objects.get_or_None(domain=domain, location_id=loc_id)
 
+    def get_assigned_location_ids(self, domain):
+        return getattr(self.get_domain_membership(domain), 'assigned_location_ids', None)
+
+    @memoized
+    def get_assigned_sql_locations(self, domain):
+        from corehq.apps.locations.models import SQLLocation
+        loc_ids = self.get_assigned_location_ids(domain)
+        if loc_ids:
+            return SQLLocation.objects.get_locations(loc_ids)
+        else:
+            return []
+
```

Comment:
haha, I actually just did the [same thing](https://github.com/dimagi/commcare-hq/commit/e85868196739e516e2b6d789b2acf54c24f90fd0) (I think you need to merge in master to get them).  Great minds...